### PR TITLE
added workaround for removing comments from rst

### PR DESCRIPTION
**Pull request by jspaaks**
**List of related issues or pull requests**

Refs: #216

**Describe the changes made in this pull request**

In this PR, I created a new directory / module `workarounds`, that is meant to contain all functionality that is either prone to change (e.g. using BeautifulSoup instead of API), not implemented (e.g. removing comments from ReStructuredText), or that has other potential problems. By collecting this functionality into one module, the workarounds are more easily identified as such, the documentation can be more organized, and any module that uses workarounds will be easily identified by just reading the `import`s. For example, I've include one such example in this PR, (at the top of Readme.py)

```python
from .workarounds import remove_comments_rst as remove_comments_with_workaround
```

and then in the body further down:

```python
def _remove_comments(self):
    if self.file_format == ReadmeFormat.MARKDOWN:
        self.text = re.sub(r"<!--.*?-->", "", self.text, flags=re.DOTALL)
    if self.file_format == ReadmeFormat.RESTRUCTUREDTEXT:
        self.text = remove_comments_with_workaround(self.text, self.filename)
    return self
```

That way it's clear what's what.

Additionally, I've renamed `schema.py`'s `include_comments` to `ignore_commented_badges` after suggestion by @fdiblen, and updated `Checker`'s getter method accordingly. 

**Instructions to review the pull request**

```shell
# checkout branch using development setup instructions
$ howfairis -d > .custom.howfairis.yml
$ nano .custom.howfairis.yml # (Ensure ignore_commented_badges: true)
$ howfairis https://github.com/fair-software/howfairis -u .custom-howfairis.yml
```

There should be a warning on stdout

### CODE REVIEW

File: howfairis/__init__.py [hunk 1]
```diff
@@ -17,6 +18,6 @@
     "cli",
     "Compliance",
     "Platform",
-    "Repo"
-
+    "Repo",
+    "workarounds"
```
Comment (sverhoeven) [inline howfairis/__init__.py:22]: Dont think workaround needs to be public. All workaounds are called directly/indirectly from a the public API.
The workarounds are internal to the library.
```suggestion
    "Repo"
```

File: howfairis/checker.py [hunk 2]
```diff
@@ -77,12 +74,8 @@ def remove_comments(text):
...
                 readme_file_format = None
 
-            if self.include_comments is True:
-                text = response.text
-            else:
-                text = remove_comments(response.text)
-
-            return Readme(filename=readme_filename, text=text, file_format=readme_file_format)
+            return Readme(filename=readme_filename, text=response.text, file_format=readme_file_format,
+                          ignore_commented_badges=not self.ignore_commented_badges)
```
Comment (sverhoeven) [inline howfairis/checker.py:78]: Huh, why invert ignore_commented_badges?
```suggestion
                          ignore_commented_badges=self.ignore_commented_badges)
```

When I set `ignore_commented_badges: false` the warning should not appear.
When I set `ignore_commented_badges: true` the warning should only appear when a README.rst is found.

Now it is the other way around.

File: howfairis/workarounds/remove_comments_rst.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import warnings
+
+
+def remove_comments_rst(text, fname):
+    warnings.warn("Using workaround to remove comments from {0}.".format(fname))
```
Comment (sverhoeven) [inline howfairis/workarounds/remove_comments_rst.py:5]: Lets use the proper warning so #72 can be closed
```suggestion
    warnings.warn("Unable to ignore comments in RestructuredText format of {0}, checks will also see comments".format(fname))
```

**Review by sverhoeven — CHANGES_REQUESTED** [review]
Nice to see the comment removal in the Readme class.

Slight typos in instructions Got output
```
.../howfairis/workarounds/remove_comments_rst.py:5: UserWarning: Using workaround to remove comments from README.rst.
  warnings.warn("Using workaround to remove comments from {0}.".format(fname))
...
```

Spun off https://github.com/fair-software/howfairis/issues/226

File: howfairis/__init__.py (hunk 1 above)
Comment (jspaaks) [reply howfairis/__init__.py:22]: Agree will change

File: howfairis/checker.py (hunk 2 above)
Comment (jspaaks) [reply howfairis/checker.py:78]: Simply a mistake on my part, I must have forgot to flip the sign after migrating from `self.include_comments`

File: howfairis/workarounds/remove_comments_rst.py (hunk 3 above)
Comment (jspaaks) [reply howfairis/workarounds/remove_comments_rst.py:5]: Good with me

**Comment by jspaaks** [comment]
I don't think I get https://github.com/fair-software/howfairis/pull/220#pullrequestreview-591339690, could you explain it?

**Comment by jspaaks** [comment]
Updated the PR instructions for convenience, sorry for the confusion

**Comment by jspaaks** [comment]
Yay! Thanks for reviewing

**Pull request merged**